src/m365/spp/commands/model/model-list.spec.ts

```diff
@@ -0,0 +1,207 @@
+        "ContentTypeGroup": "Intelligent Document Content Types",
+        "ContentTypeId": "0x010100A5C3671D1FB1A64D9F280C628D041692",
+        "ContentTypeName": "TeachingModel",
+        "Created": "2024-05-23T16:51:29Z",
+        "CreatedBy": "i:0#.f|membership|user@contoso.onmicrosoft.com",
+        "DriveId": "b!qTVLltt1P02LUgejOy6O_1amoFeu1EJBlawH83UtYbQs_H3KVKAcQpuQOpNLl646",
+        "Explanations": "{\"Classifier\":[{\"id\":\"950f39cd-5e72-4442-ae8a-ea4bae32962c\",\"kind\":\"regexFeature\",\"name\":\"Email address\",\"active\":true,\"pattern\":\"[A-Za-z0-9._%-]+@[A-Za-z0-9.-]+.[A-Za-z]{2,6}\"},{\"id\":\"d3f2940d-1df1-4ba8-975a-db3a4d626d5c\",\"kind\":\"dictionaryFeature\",\"name\":\"FirstName\",\"active\":true,\"nGrams\":[\"Michał\"],\"caseSensitive\":false,\"ignoreDigitIdentity\":false,\"ignoreLetterIdentity\":false},{\"id\":\"077966e1-73be-44c9-855f-a4eade6a280b\",\"kind\":\"modelFeature\",\"name\":\"Name\",\"active\":true,\"modelReference\":\"Name\",\"conceptId\":\"309b64e5-acd5-4538-a5b6-c6bfcdc1ffbf\"}],\"Extractors\":{\"Name\":[{\"id\":\"69e412bc-e5bc-4657-b378-34a01966bb92\",\"kind\":\"dictionaryFeature\",\"name\":\"Before label\",\"active\":true,\"nGrams\":[\"Test\",\"'Surname\",\"Test (\"],\"caseSensitive\":false,\"ignoreDigitIdentity\":false,\"ignoreLetterIdentity\":false}]}}",
+        "ID": 1,
+        "LastTrained": "2024-05-24T17:51:16Z",
+        "ListID": "ca7dfc2c-a054-421c-9b90-3a934b97ae3a",
+        "ModelSettings": "{\"ModelOrigin\":{\"LibraryId\":\"8a6027ab-c584-4394-ba9c-3dc4dd152b65\",\"Published\":false,\"SelecteFileUniqueIds\":[]}}",
+        "ModelType": 2,
+        "Modified": "2024-05-24T17:51:02Z",
+        "ModifiedBy": "i:0#.f|membership|user@contoso.onmicrosoft.com",
+        "ObjectId": "01HQDCWVGIEBDRN3RVK5A3UJW3M4TCMT45",
+        "PublicationType": 0,
+        "Schemas": "{\"Version\":2,\"Extractors\":{\"Name\":{\"concepts\":{\"309b64e5-acd5-4538-a5b6-c6bfcdc1ffbf\":{\"name\":\"Name\"}},\"relationships\":[],\"id\":\"Name\"}}}",
+        "SourceSiteUrl": "https://contoso.sharepoint.com/sites/SyntexTest",
+        "SourceUrl": null,
+        "SourceWebServerRelativeUrl": "/sites/SyntexTest",
+        "UniqueId": "164720c8-35ee-4157-ba26-db6726264f9d"
+      }
+    ];
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/web?$select=WebTemplateConfiguration`) {
+        return {
+          WebTemplateConfiguration: 'CONTENTCTR#0'
+        };
+      }
+
+      if (opts.url === 'https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models') {
+        return { value: models };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal' } });
+    assert(loggerLogSpy.calledWith(models));
+  });
+
+  it('gets correct model list when the the site URL has trailing slash', async () => {
+    const models = [
```

Comment:
Since we're using this object multiple times, let's centralize it at the top of this file.